Docs(v2): add browsers support documentation

Josh-Cena opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.

You can learn more about contributing to Docusaurus here: https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md

Happy contributing!

-->

## Motivation

To address #4751.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

slorber commented on website/docs/browser-support.md:
````diff
@@ -0,0 +1,131 @@
+samsung 13.0
+```
+
+## Accepted values {#accepted-values}
````
I think this section is unnecessary, we should just link to the browserslist doc

slorber commented on website/docs/browser-support.md:
```diff
@@ -0,0 +1,131 @@
+
+A `not` combiner performs a relative complement: `> 0.5%, not dead` eliminates the browsers that are `dead` from the ones selected by `> 0.5%`. Obviously you can _not_ start with a `not` combiner, since there is no left-hand side query to combine it with. The left-hand is always resolved as `and` combiner even if `or` is used (this is an API implementation specificity).
+
+## Choosing a value {#choosing-a-value}
```
I think this section is unnecessary, we should just link to the browserslist doc

Josh-Cena replied:
@chetstone Makes sense. I did make slight modifications to the explanation, but they're kind of interspersed in the copy-pasting. @slorber Do you think I should prune out some copy-pasted text?

slorber replied:
Not against adding more explainations, but not a direct and exhaustive copy/paste of the doc

slorber commented on website/docs/browser-support.md:
````diff
@@ -0,0 +1,131 @@
+
+You can "evaluate" any config with the `browserlist` cli to obtain the actual list:
+
+```bash
+yarn browserslist --env="production"
+```
````
```suggestion
npx browserslist --env="production"
```

This will work for both npm/yarn users without requiring browserslist in website deps

slorber commented on website/docs/browser-support.md:
```diff
@@ -0,0 +1,131 @@
+title: Browser support
+---
+
+Docusaurus allows sites to define the list of supported browsers through some [configuration](https://github.com/browserslist/browserslist).
```
```suggestion
Docusaurus allows sites to define the list of supported browsers through a [browserslist configuration](https://github.com/browserslist/browserslist).
```

slorber reviewed (changes requested):
Thanks, that looks like a nice first version to me

I don't think duplicating the config doc is worth it, a link is enough

Josh-Cena commented:
I've addressed the suggestions. However I'm still thinking about the balance between we documenting the APIs and simply providing a link to the original documentation... I think most developers who don't have the knowledge of browserslist in the first place will most likely prefer to stay on one site (i.e. the docusaurus site) instead of having to browse through the long browserslist documentation. Anyways, I've merged the last two sections as one `read more`.
